diffraction/WISH/bragg-detect/cnn/WISHDataSets.py

```diff
@@ -0,0 +1,33 @@
+import torch as tc
+import numpy as np
+import albumentations as A
+from albumentations.pytorch.transforms import ToTensorV2
+from bragg_utils import make_3d_array
+from mantid.simpleapi import Load, Rebunch
+
+
+class WISHWorkspaceDataSet(tc.utils.data.Dataset):
+    def __init__(self, ws_name):
+        ws = Load(Filename = ws_name, OutputWorkspace=ws_name)
+        self.rebunched_ws = Rebunch(InputWorkspace=ws, NBunch=3, OutputWorkspace=ws_name+"_rebunched")
```

Comment:
Can this rebunched workspace be hidden in the ADS (e.g. name start with `_`) - and be deleted after the data set has been made to free up memory?

Reply:
Rebunched workspace can be made hidden by including `storeInADS=True`. 

But inside the `BraggDetectCNN::find_bragg_peaks` method, after the inferencing, `createPeaksWorkspaceFromIndices` method requires the same workspace. However, it is possible to add a method inside `WISHWorkspaceDataSet` to free up after the Rebunched object after being consumed if you would agree?

Reply:
So the workspace can be stored in the ADS but hidden in the workspace list visible to the user by starting the workspace name with `_` - alternatively you can set `StoreInADS=False`. The latter is preferable if the user is never expected to interact with the workspace, but I don't mind which you choose. If  Rebunching took a long time we could keep it and use it is e.g. the user changes the confidence level, but it's quite quick so I don't think this is necessary!

Reply:
Thanks for pointing this out which can save some time in the example use case you have mentioned. 
I have updated the code to hide the Rebunched workspace by giving it an output name starting with "__" and code will check if it is already there and before redoing a Rebunch for the same output workspace name.